### ISSUE TITLE
color-words support

### ISSUE THREAD

**Issue by Oblomov** [jonas/tig#47]
Currently, tig has no support for color-words in diffs. In fact, starting tig wit the --color-words option fails with "No revisions match the given argument" (which is quite misleading as error message).

This should probably be consiered a long-term TODO, as it requires some heavy-duty re-engineering of the diff view, especially with respect to highlighting. Also, color-word support would require massaging the input parameters, since the word-diff option should always be turned into 'porcelain'.

**Comment by wilzbach** [comment]
Any news on this? It's been open for more than four years now ;-)

**Comment by timotheecour** [comment]
+1

**Comment by hhartz** [comment]
attempted exporting `TIG_DIFF_OPTS=--color-words` but no luck, seems inline color escapes are not supported?

+1

**Comment by SassNinja** [comment]
+1 for this feature

I've just discovered tig but without `--color-words` it's much less valuable for me than it would be with it

**Comment by avatar-lavventura** [comment]
Any improvement in this feature?

**Comment by hydrargyrum** [comment]
If I'm not mistaken, it is implemented in tig 2.5.5. I added this to my tigrc:

```
bind diff D :toggle word-diff
```

and it works like a charm when pressing `shift-d` (the separators are a bit verbose, but this is git's `plain` mode, not tig's fault).

**Comment by trygveaa** [comment]
> the separators are a bit verbose, but this is git's plain mode, not tig's fault

I don't think you can argue that it's git's fault when git has a color mode for the word diff. I, and I would assume most people, would like this way to display it, not the plain way.

**Comment by hydrargyrum** [comment]
Maybe the selected word-diff mode should be configurable in tig then. Plain mode is still useful for example if the terminal has no color support.
